protocol/app/upgrades/v.cosmwasm.0/upgrade.go

```diff
@@ -0,0 +1,164 @@
+				modAccName,
+				newModuleAccount,
+			))
+			continue
+		}
+
+		// Account has not been initialized at all. Initialize it as module.
+		// Implementation taken from
+		// https://github.com/dydxprotocol/cosmos-sdk/blob/bdf96fdd/x/auth/keeper/keeper.go#L213
+		newModuleAccount := authtypes.NewEmptyModuleAccount(modAccName, perms...)
+		maccI := (ak.NewAccount(ctx, newModuleAccount)).(sdk.ModuleAccountI) // this set the account number
+		ak.SetModuleAccount(ctx, maccI)
+		ctx.Logger().Info(fmt.Sprintf(
+			"Successfully initialized module account in state: %+v",
+			newModuleAccount,
+		))
+	}
+}
+
+func removeStatefulFOKOrders(ctx sdk.Context, k clobtypes.ClobKeeper) {
+	allStatefulOrders := k.GetAllStatefulOrders(ctx)
+	for _, order := range allStatefulOrders {
+		if order.TimeInForce == clobtypes.Order_TIME_IN_FORCE_FILL_OR_KILL {
+			// Remove the orders from state.
+			k.MustRemoveStatefulOrder(ctx, order.OrderId)
+
+			// Send indexer event for removal of stateful order.
+			k.GetIndexerEventManager().AddTxnEvent(
+				ctx,
+				indexerevents.SubtypeStatefulOrder,
+				indexerevents.StatefulOrderEventVersion,
+				indexer_manager.GetBytes(
+					indexerevents.NewStatefulOrderRemovalEvent(
+						order.OrderId,
+						indexershared.ConvertOrderRemovalReasonToIndexerOrderRemovalReason(
+							clobtypes.OrderRemoval_REMOVAL_REASON_CONDITIONAL_FOK_COULD_NOT_BE_FULLY_FILLED,
+						),
+					),
+				),
+			)
+		}
+	}
+}
+
+func initRevShareModuleState(
+	ctx sdk.Context,
+	revShareKeeper revsharetypes.RevShareKeeper,
+	priceKeeper pricetypes.PricesKeeper,
+) {
+	// Initialize the rev share module state.
+	params := revsharetypes.MarketMapperRevenueShareParams{
+		Address:         authtypes.NewModuleAddress(authtypes.FeeCollectorName).String(),
+		RevenueSharePpm: 0,
+		ValidDays:       0,
+	}
+	err := revShareKeeper.SetMarketMapperRevenueShareParams(ctx, params)
+	if err != nil {
+		panic(fmt.Sprintf("failed to set market mapper revenue share params: %s", err))
+	}
+
+	// Initialize the rev share details for all existing markets.
+	markets := priceKeeper.GetAllMarketParams(ctx)
+	for _, market := range markets {
+		revShareDetails := revsharetypes.MarketMapperRevShareDetails{
+			ExpirationTs: 0,
+		}
+		revShareKeeper.SetMarketMapperRevShareDetails(ctx, market.Id, revShareDetails)
+	}
+}
+
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	clobKeeper clobtypes.ClobKeeper,
+	revShareKeeper revsharetypes.RevShareKeeper,
+	priceKeeper pricetypes.PricesKeeper,
+	ak authkeeper.AccountKeeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx context.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		sdkCtx := lib.UnwrapSDKContext(ctx, "app/upgrades")
+		sdkCtx.Logger().Info(fmt.Sprintf("Running %s Upgrade...", UpgradeName))
+
+		InitializeModuleAccs(sdkCtx, ak)
+		// Remove all stateful FOK orders from state.
+		removeStatefulFOKOrders(sdkCtx, clobKeeper)
```

Comment:
Remove anything that's not necessary for CosmWasm upgrade

Reply:
We're upgrading from 5-> cosmwasm upgrade. Some code there is for new modules and migrations that need to be applied for 5->6. I was planning to remove this later when we are merging to main

Reply:
Given v6[ code complete](https://www.notion.so/dydx/Code-Complete-Tested-13d0fc7c14134f1dae2a2e5b269ca89d?pvs=4) is tomorrow, let's plan on upgrading from v6. I think this requires setting up the pre-upgrade with v6 binary which doesn't exist yet. For now to reduce friction let's comment out the upgrade container test for CosmWasm, and set it up later when we are ready to merge into `main`.